hopsworks-common/src/main/java/io/hops/hopsworks/common/featurestore/trainingdatasets/TrainingDatasetController.java

```diff
@@ -370,37 +474,18 @@ public TrainingDatasetDTO updateTrainingDatasetMetadata(
   
     // Verify general entity related information
     featurestoreInputValidation.verifyUserInput(trainingDatasetDTO);
-  
-    // Verify training dataset specific information
-    verifyTrainingDatasetDataFormat(trainingDatasetDTO.getDataFormat());
-  
+
     //Get jobs
     List<Jobs> jobs = getJobs(trainingDatasetDTO.getJobs(), featurestore.getProject());
     //Store jobs
-    featurestoreJobController.insertJobs(trainingDataset, jobs);
-    
+    featurestoreJobFacade.insertJobs(trainingDataset, jobs);
+    List<FeaturestoreJob> updatedJobsList = featurestoreJobFacade.getByTrainingDataset(trainingDataset);
+
     // Update metadata
-    trainingDataset.setDataFormat(trainingDataset.getDataFormat());
     trainingDataset.setDescription(trainingDatasetDTO.getDescription());
     TrainingDataset updatedTrainingDataset = trainingDatasetFacade.updateTrainingDatasetMetadata(trainingDataset);
-    // Store features
-    featurestoreFeatureController.updateTrainingDatasetFeatures(trainingDataset, trainingDatasetDTO.getFeatures());
-    switch (updatedTrainingDataset.getTrainingDatasetType()) {
-      case HOPSFS_TRAINING_DATASET:
-        hopsfsTrainingDatasetController.updateHopsfsTrainingDatasetMetadata(trainingDataset.getHopsfsTrainingDataset(),
-          (HopsfsTrainingDatasetDTO) trainingDatasetDTO);
-        break;
-      case EXTERNAL_TRAINING_DATASET:
-        externalTrainingDatasetController.updateExternalTrainingDatasetMetadata(
-          trainingDataset.getExternalTrainingDataset(), (ExternalTrainingDatasetDTO) trainingDatasetDTO);
-        break;
-      default:
-        throw new FeaturestoreException(RESTCodes.FeaturestoreErrorCode.ILLEGAL_TRAINING_DATASET_TYPE, Level.FINE,
-          ", Recognized training dataset types are: " + TrainingDatasetType.HOPSFS_TRAINING_DATASET + ", and: " +
-            TrainingDatasetType.EXTERNAL_TRAINING_DATASET + ". The provided training dataset type was not recognized: "
-            + trainingDatasetDTO.getTrainingDatasetType());
-        
-    }
+    updatedTrainingDataset.setJobs(updatedJobsList);
```

Comment:
Aren't these automatically fetched and in the `updatedTrainingDataset`?

Reply:
Nope, that object just contains what was fetched at the beginning and what has been modified. It doesn't know anything about the new jobs, that's why I re-fetch them from the database and set it manually.